module/VuFind/src/VuFind/Search/SolrCollection/Params.php

```diff
@@ -67,7 +67,7 @@ class Params extends \VuFind\Search\Solr\Params
      */
     public function initFromRecordDriver($driver, bool $hasSearch = false)
     {
-        $this->collectionID = $driver->getUniqueID();
+        $this->collectionID = $driver->getCollectionSearchId();
```

Comment:
If you move getCollectionSearchId to a trait, perhaps it would make sense to revise this to:

```
$this->collectionID = $driver->tryMethod('getCollectionSearchId') ?? $driver->getUniqueID();
```

That should maximize backward compatibility while fully supporting the new feature.